### DISCUSSION TITLE
How to protect against DDoS attacks? Rate limiting?

### DISCUSSION THREAD

**Discussion by 0xPratham** [vercel/community#1111] [Help] [answered]
I really want to know how vercel protect from ddos attacks any rate limiting or other stuff or using redis to implement leak bucket technique

**Comment by leerob** [comment, marked as answer]
Hey! We have two examples here you can check out:

- https://edge-functions-api-rate-limit.vercel.app
- https://edge-functions-api-rate-limit-and-tokens.vercel.app

**Comment by leerob** [comment]
Wanted to follow up here.

We just launched **[Attack Challenge Mode](https://vercel.com/changelog/prevent-malicious-traffic-with-attack-challenge-mode-for-vercel-firewall)** –an extra layer of defense for DDoS and malicious traffic, free for all plans. When enabled, the Vercel Firewall will serve a challenge page to help mitigate attacks.

Hope this helps!
